### ISSUE TITLE
Reversing non-empty array drops non-emptiness

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#1470]
https://psalm.dev/r/72d38b8cdc
```php
<?php

/** @return string[] */
function getOrderings(): array {
  return [];
}

$orderings = getOrderings();
if ($orderings) {
  $next = null;
  foreach (array_reverse($orderings) as $_k => $_v) {
    $next = 1;
  }
  atan($next);
}
```
**Expected**: no PossiblyNull error, as $next is set in the loop over non-empty array
**Actual**:
```
Psalm output (using commit 3704e75): 

ERROR: PossiblyNullArgument - 14:8 - Argument 1 of atan cannot be null, possibly null value provided
```
Refs https://github.com/doctrine/collections/pull/177/files/88f1f21c98309b7fce78f340e4cf68f4e8219636#r266077646
